src/Databases/DatabaseOverlay.cpp

```diff
@@ -19,13 +27,21 @@ namespace ErrorCodes
     extern const int UNKNOWN_TABLE;
 }
 
-DatabaseOverlay::DatabaseOverlay(const String & name_, ContextPtr context_)
-    : IDatabase(name_), WithContext(context_->getGlobalContext()), log(getLogger("DatabaseOverlay(" + name_ + ")"))
+DatabaseOverlay::DatabaseOverlay(const String & name_, ContextPtr context_, bool readonly_)
+    : IDatabase(name_)
+    , WithContext(context_->getGlobalContext())
+    , log(getLogger("DatabaseOverlay(" + name_ + ")"))
+    , readonly(readonly_)
 {
 }
 
 DatabaseOverlay & DatabaseOverlay::registerNextDatabase(DatabasePtr database)
 {
+    if (!database)
+        throw Exception(
+            ErrorCodes::BAD_ARGUMENTS,
```

Comment:
```suggestion
            ErrorCodes::LOGICAL_ERROR,
```

Shouldn't this be a logical error? (I hope it is already enumerated in `ErrorCodes` above)